src/include/txn_inline.h

```diff
@@ -251,23 +292,15 @@ __wt_txn_op_delete_apply_prepare_state(WT_SESSION_IMPL *session, WT_REF *ref, bo
         WT_ASSERT(session, ref->page != NULL && ref->page->modify != NULL);
         if ((updp = ref->page->modify->inst_updates) != NULL)
             for (; *updp != NULL; ++updp) {
-                (*updp)->start_ts = ts;
-                /*
-                 * Holding the ref locked means we have exclusive access, so if we are committing we
-                 * don't need to use the prepare locked transition state.
-                 */
-                (*updp)->prepare_state = prepare_state;
                 if (commit)
-                    (*updp)->durable_ts = txn->durable_timestamp;
+                    __txn_resolve_prepared_update(session, *updp);
+                else
+                    __txn_prepare_update(session, *updp);
             }
     }
-    page_del = ref->page_del;
-    if (page_del != NULL) {
-        page_del->timestamp = ts;
-        if (commit)
-            page_del->durable_timestamp = txn->durable_timestamp;
-        WT_PUBLISH(page_del->prepare_state, prepare_state);
-    }
+
+    if (ref->page_del != NULL)
```

Comment:
You could move that into the function call, I don't mind either way. I also think we should do the same shared function for regular updates not just page_del structs.

Reply:
done